Call parentNode() instead of parentOrShadowHostNode() on Node types that cannot be shadow roots

cdumez opened the pull request:
#### 9a64c570c29a4bc9021922493b4167056c2c6eef
<pre>
Call parentNode() instead of parentOrShadowHostNode() on Node types that cannot be shadow roots
<a href="https://bugs.webkit.org/show_bug.cgi?id=253776">https://bugs.webkit.org/show_bug.cgi?id=253776</a>

Reviewed by Darin Adler.

Call parentNode() instead of parentOrShadowHostNode() on Node types that cannot
be shadow roots. This avoids a shadow root check.

* Source/WebCore/dom/CharacterData.h:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/Text.cpp:
(WebCore::isSVGShadowText):
(WebCore::isSVGText):
(WebCore::Text::createTextRenderer):
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::ownerSVGElement const):
(WebCore::SVGElement::viewportElement const):

Canonical link: <a href="https://commits.webkit.org/261601@main">https://commits.webkit.org/261601@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2cd4645667fc9c065e5d72879a1614d7b1312e43

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112260 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21398 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/891 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4055 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/120893 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116324 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22741 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/12466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/4829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118023 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16900 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100080 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105319 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/98857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/45906 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/13791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/664 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/94241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14475 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/12466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/19834 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52667 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8089 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16272 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/dom/CharacterData.h:
```diff
@@ -43,6 +43,8 @@ class CharacterData : public Node {
     // Like appendData, but optimized for the parser (e.g., no mutation events).
     void parserAppendData(StringView);
 
+    ContainerNode* parentOrShadowHostNode() const = delete; // Call parentNode() instead.
```
I sometimes make the return value `void` and make these private as well.

cdumez replied:
Ok.

darinadler reviewed (approved):
What is the common thing between which these are? What about `DocumentFragment`?

cdumez committed:
Call parentNode() instead of parentOrShadowHostNode() on Node types that cannot be shadow roots
https://bugs.webkit.org/show_bug.cgi?id=253776

Reviewed by Darin Adler.

Call parentNode() instead of parentOrShadowHostNode() on Node types that cannot
be shadow roots. This avoids a shadow root check.

* Source/WebCore/dom/CharacterData.h:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/Text.cpp:
(WebCore::isSVGShadowText):
(WebCore::isSVGText):
(WebCore::Text::createTextRenderer):
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::ownerSVGElement const):
(WebCore::SVGElement::viewportElement const):

Canonical link: https://commits.webkit.org/261601@main
